### ISSUE TITLE
Possible to prevent saving of Credential Password property

### ISSUE THREAD

**Issue by alwortma** [TabularEditor/TabularEditor3#479]
I would like to be able exclude entirely the Password property of a credential when "Include Sensitive" is not enabled in Model -> Serialization Options.

Currently (TE3 v3.2.2) if the Model.bim file **has ever been opened with a workspace** it saves the Datasource with the password property set to a literal string of asterisks.

```
      {
        "type": "structured",
        "name": "Teradata",
        "connectionDetails": {
          "protocol": "teradata",
          "address": {
            "server": "[server_url]"
          },
          "authentication": null,
          "query": null
        },
        "options": {
          "hierarchicalNavigation": true
        },
        "credential": {
          "AuthenticationKind": "UsernamePassword",
          "Username": "[username]",
          "EncryptConnection": false,
          "PrivacySetting": "Organizational",
          "Password": "********"
        }
      }
```

If "Include sensitive" is enabled, this string of asterisks is replaced with the actual password.

Ideally, if that option is **_not enabled_**, I would want the property value to just be dropped.

```
    {
        "type": "structured",
        "name": "Teradata",
        "connectionDetails": {
          "protocol": "teradata",
          "address": {
            "server": "[server_url]"
          },
          "authentication": null,
          "query": null
        },
        "options": {
          "hierarchicalNavigation": true
        },
        "credential": {
          "AuthenticationKind": "UsernamePassword",
          "Username": "[username]",
          "EncryptConnection": false,
          "PrivacySetting": "Organizational"
        }
      }
```

**Comment by otykier** [comment]
Good catch, thanks for reporting (interestingly, this is the default behavior of TOM... however, we should be able to modify the JSON before it is saved)

**Comment by otykier** [comment]
Fixed in [3.2.3](https://github.com/TabularEditor/TabularEditor3/releases/tag/3.2.3). Thanks for reporting!
